src/main/java/org/terasology/caves/CaveFacetProvider.java

```diff
@@ -35,233 +36,36 @@
+            SubSampledNoise3D sNoise = new SubSampledNoise3D(bNoise, new Vector3f(0.01f, 0.01f, 0.01f), 4);
+            caveNoise[i] = sNoise;
+        }
     }
 
     @Override
     public void process(GeneratingRegion region) {
-        float width = configuration.width;
-        float height = configuration.height;
-        float gradualIncreaseOverDepth = configuration.gradualIncreaseOverDepth;
-        float amountOfCavesNearSurface = configuration.amountofCavesNearSurface;
-        float sharpSurfaceCutoffDepth = 12;
-        float minDepth = 0;
-        float amountOfCaves = configuration.amountOfCaves;
-        float noiseLevel = configuration.rawAmount;
 
-        // at default settings,  make caves wider than tall.
-        SubSampledNoise caveNoise = new SubSampledNoise(new RidgedNoise(baseCaveNoise, 2), new Vector3f(0.06f * (1f / width), 0.09f * (1f / height), 0.06f * (1f / width)), 4);
-        SubSampledNoise fadeCaveNoise = new SubSampledNoise(baseFadeCaveNoise, new Vector3f(0.006f * (1f / width), 0.006f * (1f / height), 0.006f * (1f / width)), 1);
         CaveFacet facet = new CaveFacet(region.getRegion(), region.getBorderForFacet(CaveFacet.class));
-        SurfaceHeightFacet surfaceHeightFacet = region.getRegionFacet(SurfaceHeightFacet.class);
 
         // get noise in batch for performance reasons.  Getting it by individual position takes 10 times as long
-        float[] caveNoiseValues = caveNoise.noise(facet.getWorldRegion());
-        float[] fadeCaveNoiseValues = fadeCaveNoise.noise(facet.getWorldRegion());
+        float[][] caveNoiseValues = new float[][]{caveNoise[0].noise(facet.getWorldRegion()),caveNoise[1].noise(facet.getWorldRegion())};
 
         for (Vector3i pos : region.getRegion()) {
-            float depth = surfaceHeightFacet.getWorld(pos.x, pos.z) - pos.y;
-            if (depth > minDepth) {
-                float noiseValue = caveNoiseValues[facet.getWorldIndex(pos)];
-                // fade caves out as they reach the surface or above the surface
-                float fadeForSurfaceCutoff = Math.min(1f - amountOfCavesNearSurface, Math.max(0f, 1f - (depth / sharpSurfaceCutoffDepth)));
-                // gradually decrease caves as they get closer to the surface
-                float fadeForScale = Math.max(0f, 1f - (depth / gradualIncreaseOverDepth)) * (1f - amountOfCavesNearSurface);
-
-                float noiseLevelIncrease = (1f - noiseLevel)
-                        * (
-                        Math.max(fadeForSurfaceCutoff, fadeForScale)
-                                // fade caves on a broad scale to stop them from being uniform
-                                // Amount added to the noise value: 1 = prevent all caves.  0 = allow normal perlin.  -1 = all caves
-                                + Math.max(0f, Math.abs(fadeCaveNoiseValues[facet.getWorldIndex(pos)]) + (2f * (1f - amountOfCaves)) - 1f)
-                );
+            int i = facet.getWorldIndex(pos);
```

Comment:
I'd highly appreciate an engine PR adding Javadoc for `BaseFacet3D::getWorldIndex`. Doc for the `CaveFacet::getWorldIndex` part can be added in this PR.

Reply:
I was also a little confused when I came across this in the original version. I've added a brief description, but the property that's really important here is a relation between CaveFacet.getWorldIndex and SubSampledNoise.noise(Region3i), so it's not clear where that documentation would go.